pkcs12/Cargo.toml

```diff
@@ -14,6 +14,20 @@ readme = "README.md"
 edition = "2021"
 rust-version = "1.65"
 
+[features]
+alloc = []
```

Comment:
No need to feature-gate `alloc`. This crate will definitely have a hard dependency on `alloc` by way of `cms`.

Reply:
That said, feature-gating `kdf` might make sense. You could make the `digest` dependency optional that way.

Reply:
I have added a new feature flag `kdf` for the kdf implementation and `digest` depends on that.
However, I had to put the `extern crate alloc` behind the same feature flag to make to linter happy. It could be removed once other functionality have been added to the crate.